contracts/nois-drand/src/state.rs

```diff
@@ -67,15 +67,15 @@ pub struct StoredSubmission {
 
 /// Stores the submission for an efficient (round, address) lookup
 /// An entry of this map looks like (round, drand_bot_addr) =>  time
-pub const SUBMISSIONS: Map<(u64, &Addr), StoredSubmission> = Map::new("submissions");
+pub const SUBMISSIONS: Map<(u64, &[u8]), StoredSubmission> = Map::new("s");
 
 /// A map from (round, index) to bot address. This is used when
 /// sorted submissions are needed.
 ///
 /// The `index` values are 0-based. So the `n`th submission has index
 /// n-1 here as well as in the response array in `SubmissionsResponse`.
 /// An entry of this map looks like (round, 1) =>  drand_bot_addr ; Second fastest bot
-pub const SUBMISSIONS_ORDER: Map<(u64, u32), Addr> = Map::new("submissions_order");
+pub const SUBMISSIONS_ORDER: Map<(u64, u16), CanonicalAddr> = Map::new("so");
```

Comment:
do you think it could have overlapping iterator issues ? `s` and `so`, perhaps add an extra letter or different one.

Reply:
If that is the case I'd consider it a bug in storage-plus. It's possible, not sure. It should be length-prefixed such that this cannot happen.

In general I only want to merge minor changes from this PR. I'm not convinced going to binary addresses is the right way.

Reply:
You are right I forgot about storage-plus.